src/util.py

```diff
@@ -260,12 +265,127 @@ def xr_add_cyclic_longitudes(da, coord):
 
     return new_da
 
+
+def plotCLabels(da, contours, transform, ax, proj, Clevels=[], lowClevels=[], highClevels=[], cfs=14, lfs=22, hfs=22):
+
+    """
+    Utility function to plot contour labels. Regular contour labels will be plotted using the built-in matplotlib
+    clabel function. High/Low contour labels will be plotted using text boxes for more accurate label values 
+    and placement.
+
+    Args:
+
+        da: (:class:`xarray.DataArray`):
+            Xarray data array containing the lat, lon, and field variable data values.
+
+        contours (:class:`cartopy.mpl.contour.GeoContourSet`):
+            Contour set that is being labeled.
+
+        transform (:class:`cartopy._crs`):
+            Projection that the input coordinates (in GPS form of lon, lat) should be transformed to
+            (ex. ccrs.Geodetic()).
+
+        ax (:class:`matplotlib.pyplot.axis`):
+            Axis containing the contour set.
+
+        proj (:class:`cartopy._crs`):
+            Projection of axis.
+
+        Clevels (:class:`list`):
+            List of coordinate tuples in GPS form (lon in degrees, lat in degrees)
+            that specify where the contours with regular field variable values should be plotted.
+
+        lowClevels (:class:`list`):
+            List of coordinate tuples in GPS form (lon in degrees, lat in degrees)
+            that specify where the contours with low field variable values should be plotted.
+
+        highClevels (:class:`list`):
```

Comment:
I'm a bit confused by `Clevels` `lowClevels` and `highClevels`. These arguments seem to assume that the user knows where the lows and highs will be, but that makes this function hard to use if they are plotting the data for the first time. Can you explain to me what your intention was with these arguments?

Reply:
The user will know where the low and high contour levels will be, because they will have found those values with findLocalExtrema(). This function is solely for setting the format of the labels and plotting them at the right locations.